### PR TITLE
Add a userscript to integrate with taskwarrior.

### PR DESCRIPTION
**Pull request by rcorre**
The `taskadd` userscript adds a task based on the current title and URL.
It passes additional arguments along to `task`.

For example:
    :spawn --userscript taskadd due:eod pri:H
will add a task with high priority due at the end of the day.
The description will be the current page title and it will be annotated
with the current page url.

If used with hints, the hint text is used as the description.

Unfortunately, there is currently no way to use :hint fill and maintain
the hint text, which limits the ability to provide additional args in
hint mode.

### CODE REVIEW

File: misc/userscripts/taskadd [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# use either the current page title or the hint text as the task description
+[[ $QUTE_MODE == 'hints' ]] && title=$QUTE_SELECTED_TEXT || title=$QUTE_TITLE
+
+# try to add the task and grab the output
+msg="$(task add $title $@ 2>&1)"
+
+if [[ $? == 0 ]]; then
+    # scan the output for the new task number
+    num=$(echo $msg | sed -r 's/Created task ([0-9]+)\./\1/')
```
Comment (The-Compiler) [inline misc/userscripts/taskadd]: I think you can simply use the virtual tag `+LATEST` instead of parsing the output

**Comment by The-Compiler** [comment]
Nice! I started to use taskwarrior just this week :laughing: 

> Unfortunately, there is currently no way to use :hint fill and maintain
the hint text

So you're looking for something like `{hint-title}`, similar to `{hint-url}` basically?

**Comment by lamarpavel** [comment]
Very nice, I've been missing integration of taskwarrior in other applications. Now the next step would be to open all URLs as tabs in qutebrowser that match a criteria (eg +TODAY) and were added using this script. To make filtering easier, you might want to add a tag like "qute" to every entry added by your script.

Also nice: A startup command in qutebrowser, that opens overdue tasks as tabs - just like tabs from previous sessions can be resumed.

File: misc/userscripts/taskadd (hunk 1 above)
Comment (rcorre) [inline misc/userscripts/taskadd]: Neat! Didn't know that.

**Comment by rcorre** [comment]
> So you're looking for something like {hint-title}, similar to {hint-url} basically?

Yeah, though I would have to tweak my script a little to always use the argument passed to it rather than `$QUTE_TITLE`

> you might want to add a tag like "qute" to every entry added by your script.

I tried to leave this open to the user. Additional args are passed along to `task`, so you could create a mapping to 

    :spawn --userscript taskadd +qute

> Now the next step would be to open all URLs as tabs in qutebrowser 

That would be cool, maybe as a separate userscript `taskopen`, where `taskopen <criteria>` runs `task list <criteria>` and tries to open any annotation that is a valid url.

**Comment by The-Compiler** [comment]
@rcorre I think this is ready to merge, though I'd still prefer using `+LATEST` instead of the parsing - can you give that a try?

**Comment by rcorre** [comment]
Yup, thanks for the reminder. I was gonna play with @lamarpavel's idea of a `taskopen` script but that could be a separate PR. Also maybe adding `{hint-title}` to better support using the script with hints, but at least for me that's not a common use case.

**Comment by The-Compiler** [comment]
Sorry for the delay here, going to merge this now :wink:

FWIW I got the idea of a `:print --pdf` because of this which was then implemented by @jdkaplan in #1639 :tada: That should make it possible to add a nice alias adding a "to read" todo in taskwarrior and storing the PDF of the page to read :wink:

**Comment by rcorre** [comment]
Nice! I stumbled upon [taskopen](https://github.com/ValiValpas/taskopen), which alows you open files and urls stored as annotations. I haven't tried it yet but it sounds like it could complement this nicely.

**Pull request merged**